### PR TITLE
Add language around multiple schemas; update example

### LINKED ISSUE #950: Verification of multiple credential schemas
In the [Data Schema section](https://www.w3.org/TR/vc-data-model-2.0/#data-schemas), the `credentialSchema` definition contains the following statement:

> The value of the `credentialSchema` [property](https://www.w3.org/TR/vc-data-model-2.0/#dfn-property) MUST be one or more data schemas that provide [verifiers](https://www.w3.org/TR/vc-data-model-2.0/#dfn-verifier) with enough information to determine if the provided data conforms to the provided schema.

If more than one schema is provided, then verifiers should probably check the credential against every schema provided rather than making sure that at least one of the schemas provided conforms because it would allow for composability of schemas, for example, a credentialSchema property could point to `https://vc-schema.example.org/UniversityDegree` and `https://vc-schema.example.org/NonTransferable`.

Maybe the statement could be altered slightly to make the intended processing of multiple credential schemas (all valid vs at least one valid) clearer:

> The value of the `credentialSchema` [property](https://www.w3.org/TR/vc-data-model-2.0/#dfn-property) MUST be one or more data schemas that provide [verifiers](https://www.w3.org/TR/vc-data-model-2.0/#dfn-verifier) with enough information to determine if the provided data conforms. Each `credentialSchema` MUST match the provided data for it to be determined as conforming by verifiers.

### PR DESCRIPTION
**Pull request by decentralgabe**
Fix #950
- Updated reference name to include the short name (`vc-json-schema`)
- Revised language to note processing of multiple schemas should be handled according to each specific type's rules
- Updated example to show multiple schemas (there are now examples with one and more than one schema)

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/pull/1198.html" title="Last updated on Jul 17, 2023, 6:25 PM UTC (a40c3ce)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/1198/3a0e423...a40c3ce.html" title="Last updated on Jul 17, 2023, 6:25 PM UTC (a40c3ce)">Diff</a>

### CODE REVIEW

File: index.html
Comment (msporny) [inline index.html:2448]:
```suggestion
If multiple schemas are present, validity is determined according to the
```

If we put a MUST here, we will have to test it in the core data model test suite... it's do-able, but puts a pretty high burden on this feature... and if we don't get enough implementations for this feature, it'll be struck from the specification. Are we sure that we'll have enough implementers that implement two types of JSON Schema checking for this to survive CR?

File: index.html
Comment (msporny) [inline index.html:2494]: I know I owe you a response in the other repo on the difference between these two things. I'm still concerned that we don't need both of them, but we'd need to hash this out over there. So, I won't block this PR because of this... just noting that I'm still super uneasy about having two different types of schema.

File: index.html
Comment (msporny) [inline index.html:2485]: Move this "university" block inside of the "degree" credential, otherwise you're not tying the degree to the university (and vice-versa).

**Review by msporny — CHANGES_REQUESTED** [review]
Minor changes requested, some concerns noted, but none of it is blocking.

**Review by TallTed — APPROVED** [review]
lgtm, given @msporny's suggestions

File: index.html
Comment (decentralgabe) [reply index.html:2494]: I'm fine to change these examples to just use `JsonSchema2023`, though I was hoping to demonstrate that the types do not need to be the same. Is there another second type I could use in place?

File: index.html
Comment (decentralgabe) [reply index.html:2448]: yes, we have the implementers. I'm fine with removing the MUST

**Review by andresuribe87 — APPROVED** [review]

File: index.html
Comment (decentralgabe) [reply index.html:2485]: I believe that was my intention--I want to show that there can be related data in the same credential, covered by two distinct JSON schemas.

The tying happens when the issuer signs over the entire subject IMO. 

For example, the first schema for the degree may be....
```json
{
    "$id": "https://example.com/schemas/degree-credential-schema.json",
    "$schema": "https://json-schema.org/draft/2020-12/schema",
    "name": "Degree VC Schema",
    "type": "object",
    "properties": {
      "credentialSubject": {
        "type": "object",
        "properties": {
          "degree": {
            "type": "object",
            "properties": {
              "type": { "type": "string"},
              "name": { "type": "string"}
            },
             "required": ["type", "name"]
          }
        },
        "required": ["degree"]
      }
    }
```

And the second, for the university...
```json
{
    "$id": "https://example.com/schemas/degree-credential-schema.json",
    "$schema": "https://json-schema.org/draft/2020-12/schema",
    "name": "University VC Schema",
    "type": "object",
    "properties": {
      "credentialSubject": {
        "type": "object",
        "properties": {
          "university": {
            "type": "object",
            "properties": {
              "type": { "type": "string"},
              "name": { "type": "string"}
            },
             "required": ["type", "name"]
          }
        },
        "required": ["university"]
      }
    }
```

However, if the university block was inside degree it would be awkward to have this represented by two schemas, though possible.

Comment (OR13) [reply index.html:2485]: I agree with @msporny ... however... this kinda highlights why just doing JSON-LD is not enough... 

You need to *think* about the RDF graph the JSON-LD becomes... in order to ensure that claims attribute to the correct `id`.... out spec says a bunch of informative stuff about this, which implementers ignore (because they are not required to understand the RDF graph)... but then we see a lot of comments like this... all over the place on implementations and profiles of the spec... it is a hint at a missing normative requirement.

You current example says:

`did:example:ebfeb1f712ebc6f1c276e12ec21` has a `degree`
`did:example:ebfeb1f712ebc6f1c276e12ec21` has a `university`

I guess if `did:example:ebfeb1f712ebc6f1c276e12ec21` is some identifier for a digital diploma this makes sense... so the example is valid... the comments on architecture of claims still hint at potentially providing better guidance, normatively... without some normative reason to follow them I think its safe to ignore them.

Comment (OR13) [reply index.html:2485]: Screen Shot 2023-07-12 at 5 08 29 PM

Comment (decentralgabe) [reply index.html:2485]: Yeah, after reflection, I better understand the point @msporny is making, but also agree with you @OR13 it seems like there needs to be normative guidance around this. Without it, I can see a credential being interpreted in multiple different ways.

As I have it written, it doesn't make much sense when viewed as an RDF graph. I'm wondering if there's another example I can use with separate top-level claims. Maybe both a degree and a name credential would be less controversial.

Comment (OR13) [reply index.html:2485]: Another version of the same confusion can be found here: https://github.com/w3c/vc-data-model/pull/1172#discussion_r1261786118

Comment (dlongley) [reply index.html:2485]: I think the main difference is that the VCDM requires the connections between things to be explicit in how the modeling occurs in the data vs. just being in the head of the author. You don't have to guess what the author intended for how things "could be interpreted", but rather, it's in the data itself based on the required patterns. And, it's more of a consequence of that -- that we can build tools that will show the relationships between the data as a graph of information. You can't do that when the "interpretation" is in the author's head and not the structure of the data itself.

File: index.html
Comment (msporny) [inline index.html:2485]:
```suggestion
    "name": {
      "firstName": "John",
      "lastName": "Doe"
```

Example above doesn't follow best practices: https://schema.org/name

See the example here (JSON-LD tab, for now `name` is typically used): https://schema.org/Person#eg-0001

... when in doubt, look at schema.org for examples. You could do `alumniOf` (or `memberOf`):

https://schema.org/alumniOf

```
  "alumniOf": {
    "name": "Example University"
   }
```

Comment (decentralgabe) [reply index.html]: thanks, updated to `alumniOf`

File: index.html
Comment (decentralgabe) [inline index.html:2492]:
```suggestion
    "id": "https://example.org/examples/person.json",
```

File: index.html
Comment (msporny) [reply index.html]: > Is there another second type I could use in place?

At this point in time, no. :)

Having both be JsonSchema2023 would be fine, IMHO... as long as the files referenced are different (which they would be).

File: index.html
Comment (msporny) [inline index.html:2443]:
```suggestion
<code>JsonSchema2023</code>), and an <code>id</code> <a>property</a>
```

Comment (msporny) [inline index.html:2443]:
```suggestion
<code>JsonSchema2023</code>), and an <code>id</code> <a>property</a>
```

File: index.html
Comment (msporny) [inline index.html:2489]:
```suggestion
    "type": "JsonSchema2023"
```

**Review by msporny — APPROVED** [review]
Two more minor nits, take 'em or leave 'em, otherwise, LGTM.

File: index.html
Comment (brentzundel) [inline index.html:2439]: curious why the normative downgrade. Is there a use case that can't happen if the value of `credentiSchema` property MUST be one or more data schemas?

Comment (decentralgabe) [reply index.html:2439]: this was a suggestion from @TallTed's comment here https://github.com/w3c/vc-data-model/issues/950#issuecomment-1608008708


> I also note that the current VCDM spec says that `credentialSchema` "_can_ be included by an [issuer](https://www.w3.org/TR/vc-data-model-2.0/#dfn-issuers) in the [verifiable credentials](https://www.w3.org/TR/vc-data-model-2.0/#dfn-verifiable-credential) that it issues". This optionality should be reflected in the definition that follows, i.e., —
> 
> _If present_, the value of the _optional_ `credentialSchema` [property](https://www.w3.org/TR/vc-data-model-2.0/#dfn-property) MUST be one or more data schemas that provide [verifiers](https://www.w3.org/TR/vc-data-model-2.0/#dfn-verifier) with enough information to determine if the provided data conforms to the provided schema.
> 
> All things considered, it would probably also be better to change that MUST to a SHOULD.


tl;dr if including a `credentialSchema` is a _can_ then the normative guidance is a _should_.

I personally do not see a problem with leaving the language as it was, but I defer to those more experienced than me with normative language.

Comment (brentzundel) [reply index.html:2439]: with other optional properties we followed the pattern of saying the property MAY be present, but if it is present, it MUST follow the guidance. See also [validity period](https://w3c.github.io/vc-data-model/#validity-period), [status](https://w3c.github.io/vc-data-model/#status), and `id`, `verifiableCredential`, and `proof` in [presentations](https://w3c.github.io/vc-data-model/#presentations-0)

Softening the MUST to a SHOULD makes the presence of the property not mean the same thing in all uses, which is substantially different than the previous wording, and it no longer matches the language in the rest of the spec. If there is no use case driving this change, I am opposed to it.

Comment (OR13) [reply index.html:2439]: It also means we don't need to test it... MUST requires us to test, SHOULD is optional to test.

Comment (decentralgabe) [reply index.html:2439]: OK, reading this I am in favor of reverting my change. Reverted.

Comment (decentralgabe) [inline index.html:2439]:
```suggestion
The value of the <code>credentialSchema</code> <a>property</a> MUST be one or
```

**Review by brentzundel — APPROVED** [review]

**Review by dlongley — APPROVED** [review]

**Comment by decentralgabe** [comment]
All comments addressed. Been over a week. Multiple editor approvals. Merging

**Pull request merged**
